app/src/main/java/no/ntnu/klubbhuset/ui/userviews/club/detailed/ClubDetailedFragment.java

```diff
@@ -66,51 +63,47 @@ public void onAttach(Context context) {
-
-        // decide what content to show based on membership status
-        mViewModel.getMembership(club).observe(this, response -> {
-            if (response.getStatus() == Status.SUCCESS) {
-                mListener.onMembershipStatusChanged(response.getData());
-            } else {
-                //todo handle error
-            }
-        });
-
+        mViewModel = ViewModelProviders.of(getActivity()).get(ClubDetailedViewModel.class);
 
         name = getView().findViewById(R.id.club_detailed_name);
         description = getView().findViewById(R.id.club_detailed_description);
         url = getView().findViewById(R.id.club_detailed_homepage);
         email = getView().findViewById(R.id.club_detailed_email);
         image = getView().findViewById(R.id.club_detailed_banner);
-        if (club.getImage() == null || club.getImage().isEmpty()) {
+
+        mViewModel.getCurrentClub().observe(this, clubResource -> {
+            if (clubResource.getStatus() == Status.SUCCESS) {
+                this.club = clubResource.getData();
+                setData();
+            } else if (clubResource.getStatus() == Status.LOADING){
+                // loading
+            } else {
+                Toast.makeText(getContext(), "Error loading club", Toast.LENGTH_SHORT).show();;
+            }
+        });
+    }
+
+    private void setData() {
+        if (club.getOrgImages() == null || club.getOrgImages().length == 0) {
             // set placeholder
-            image.setImageResource(R.drawable.ic_landscape_black_24dp);
+            image.setImageResource(R.drawable.ic_broken_image_black_24dp);
         } else {
-            String encodedString = club.getImage();
-
-            byte [] encodeByte = Base64.decode(encodedString, Base64.DEFAULT);
-            Bitmap bitmap = BitmapFactory.decodeByteArray(encodeByte, 0, encodeByte.length);
-            image.setImageBitmap(bitmap);
+            image.setImageBitmap(club.getOrgImages()[0].getImage());
         }
 
         name.setText(club.getName());
         description.setText(club.getDescription());
         url.setText(club.getUrl());
         email.setText(club.getEmailContact());
 
-        /*joinClubBtn = getView().findViewById(R.id.club_detailed_joinbtn);
-        joinClubBtn.setOnClickListener(click -> {
-            mViewModel.joinClub(club.getOid()).observe(this, response -> {
-                if (response != null) {
-                    System.out.println("what it do . . .");
-                    mListener.onMembershipStatusChanged(response);
-                } else {
-
-                }
-            });
-        });*/
+        // decide what content to show based on membership status
+        mViewModel.getMembership(club).observe(this, response -> {
+            if (response.getStatus() == Status.SUCCESS) {
+                mListener.onMembershipStatusChanged(response.getData());
+            } else {
+                //todo handle error
+            }
```

Comment:
Burde det hær vær inkludert i den her pull requesten? 100 push-ups

Reply:
måtte legge til for å ikkje krasje appen. 200 push-ups